demisto_sdk/commands/init/initiator.py

```diff
@@ -720,45 +912,100 @@ def script_init(self) -> bool:
 
-        from_version = input("\nThe fromversion value that will be used (optional): ")
-        if from_version:
+        yml_dict["id"] = id_from_yml.replace(self.HELLO_WORLD_PACK_NAME, self.dir_name)
+        yml_dict["name"] = name_from_yml.replace(
+            self.HELLO_WORLD_PACK_NAME, self.dir_name
+        )
+
+        content_item = "modeling rules" if self.is_modeling_rules else "parsing rules"
+        if from_version := input(
+            f"\nThe fromversion value that will be used for {content_item} (optional): "
+        ):
             yml_dict["fromversion"] = from_version
 
         if Version(
-            yml_dict.get("fromversion", DEFAULT_CONTENT_ITEM_FROM_VERSION)
-        ) < Version(self.SUPPORTED_FROM_VERSION):
-            yml_dict["fromversion"] = self.SUPPORTED_FROM_VERSION
-
-        if integration:
-            yml_dict["display"] = self.id
-            yml_dict["category"] = (
-                self.category
-                if self.category
-                else Initiator.get_valid_user_input(
-                    options_list=INTEGRATION_CATEGORIES,
-                    option_message="\nIntegration category options: \n",
-                )
-            )
-
+            yml_dict.get("fromversion") or DEFAULT_CONTENT_ITEM_FROM_VERSION
+        ) < Version(self.SUPPORTED_FROM_VERSION_XSIAM):
+            yml_dict["fromversion"] = self.SUPPORTED_FROM_VERSION_XSIAM
         with open(
             os.path.join(self.full_output_path, f"{self.dir_name}.yml"), "w"
         ) as f:
             yaml.dump(yml_dict, f)
 
-        os.remove(os.path.join(self.full_output_path, f"{current_suffix}.yml"))
+    def yml_reformatting(self, current_suffix: str, integration: bool = False):
+        """Formats the given yml to fit the newly created integration/script
+
+        Args:
+            current_suffix (str): The yml file name (HelloWorld or HelloWorldScript)
+            integration (bool): Indicates if integration yml is being reformatted.
+        """
+        yml_dict = get_yaml(
+            os.path.join(self.full_output_path, f"{current_suffix}.yml")
+        )
+        if yml_dict:
+            yml_dict["commonfields"]["id"] = (
+                f"{self.id}{EVENT_COLLECTOR}"
+                if self.xsiam and self.is_integration
+                else self.id
+            )
+            yml_dict["name"] = (
+                f"{self.id}{EVENT_COLLECTOR}"
+                if self.xsiam and self.is_integration
+                else self.id
+            )
+
+            from_version = input(
+                "\nThe fromversion value that will be used (optional): "
+            )
+            if from_version:
+                yml_dict["fromversion"] = from_version
+
+            if not self.xsiam and Version(
+                yml_dict.get("fromversion", DEFAULT_CONTENT_ITEM_FROM_VERSION)
+            ) < Version(self.SUPPORTED_FROM_VERSION):
+                yml_dict["fromversion"] = self.SUPPORTED_FROM_VERSION_XSIAM
```

Comment:
Are you sure this is correct?
Also, it is not the code in your branch. Make sure the PR is updated please